### PR TITLE
Edits to advanced Bash chapter

### PR DESCRIPTION
**Pull request by gvwilson**
Please proof-read; there is one FIXME that needs checking.

Depends on #383.

### CODE REVIEW

File: py-rse/bash-advanced.Rmd [hunk 1]
```diff
@@ -254,36 +258,40 @@ $ cat summarize_all_books.sh
 302 history 6 > summarize_all_books.sh
 ```
 
-Open the file in an editor,
+We can now open the file in an editor,
```
Comment (cwickham) [inline py-rse/bash-advanced.Rmd:261]: Isn't this still a hypothetical?
```suggestion
We could now open the file in an editor,
```
Which begs the question, why don't we just get the learner to run those six lines, then it doesn't have to be hypothetical...

File: py-rse/bash-advanced.Rmd [hunk 2]
```diff
@@ -505,28 +504,29 @@ The distinction is clear. For example, you have frequently seen
...
 It also has a lot of options,
-but unlike most Unix commands these are written as full words rather than abbreviations.
+but unlike most Unix commands they are written as full words
+rather than single-letter abbreviations.
 To show how it works,
-we will use the entire contents of our `zipfs_law` directory,
+we will use the entire contents of our `zipf` directory,
 including files we created earlier in this chapter:
 
 ├── book_summary.sh
```
Comment (cwickham) [inline py-rse/bash-advanced.Rmd:532]: Needs a `` ```text``, and appropriate closing

File: py-rse/bash-advanced.Rmd [hunk 3]
```diff
@@ -625,10 +625,31 @@ $ find . -name "*.txt"
...
+```
+
+and gives us the error message:
+
+```text
+find: -name: requires additional arguments
+```
+
+We have seen before how to combine commands using pipes.
+let's use another technique to see how large our books are:
```
Comment (cwickham) [inline py-rse/bash-advanced.Rmd:652]:
```suggestion
We have seen before how to combine commands using pipes,
let's use another technique to see how large our books are:
```

File: py-rse/bash-advanced.Rmd (hunk 1 above)
Comment (cwickham) [reply py-rse/bash-advanced.Rmd:261]: Actually later output includes `authors.txt`, `releases.txt`, and `summarize_all_books.sh`, so we should definitely make learners run these commands.

File: py-rse/bash-advanced.Rmd [hunk 4]
```diff
@@ -818,17 +849,19 @@ $ echo $HOME
...
-as we have seen,
-`echo` simply prints the literal string `HOME` if we omit it
-(because that's what we want `echo` to do if we type `echo hello` or something similar).
-This works just like wildcards:
-the shell does the replacement before running the program we've asked for.
-Thanks to this expansion, what we actually run is `echo /Users/amira`,
+As with loop variables (Section \@ref(py-rse-bash-basics-loops)),
+the dollar sign before the variable names tells the shell
+that we want the variable's value.
+This works just like wildcards expansion
```
Comment (cwickham) [inline py-rse/bash-advanced.Rmd:855]:
```suggestion
This works just like wildcard expansion,
```

File: py-rse/bash-advanced.Rmd [hunk 5]
```diff
@@ -869,14 +897,27 @@ export BACKUP_DIR=$TEMP_DIR/backup
 These three lines create the variables `DEPARTMENT`,
 `TEMP_DIR`,
 and `BACKUP_DIR`,
-and export them so that any programs the shell runs can see them as well.
+and [export][export-variable] them so that any programs the shell runs can see them as well.
```
Comment (cwickham) [inline py-rse/bash-advanced.Rmd:900]: Missing link?

File: py-rse/bash-advanced.Rmd [hunk 6]
```diff
@@ -869,14 +897,27 @@ export BACKUP_DIR=$TEMP_DIR/backup
...
 Notice that `BACKUP_DIR`'s definition relies on the value of `TEMP_DIR`,
 so that if we change where we put temporary files,
 our backups will be relocated automatically.
+However,
+this will only happen once we restart the shell,
+because `.bashrc` is only executed when the shell starts up.
+
+> **What's in a Name?**
+>
+> The '.' character at the front prevents `ls` from listing this file
```
Comment (cwickham) [inline py-rse/bash-advanced.Rmd:910]:
```suggestion
> The '.' character at the front of `.bashrc` prevents `ls` from listing this file
```

File: py-rse/bash-advanced.Rmd [hunk 7]
```diff
@@ -903,49 +944,51 @@ alias rm='rm -iv'
...
-Demonstrating concepts in this chapter has resulted
-in a number of extraneous files that we won't use anymore.
-We can use these files to explore how our aliases work in practice,
-while cleaning up our project in preparation for our next chapter.
+## Summary {#py-rse-bash-advanced-summary}
+
+But as powerful as it the Unix shell is,
+it does have its shortcomings:
+dollar signs, quotes, and other punctuation
+can make a complex shell script look as though it was created by a cat dancing on a keyboard,
```
Comment (cwickham) [inline py-rse/bash-advanced.Rmd:952]:
```suggestion
can make a complex shell script look as though it was created by a cat dancing on a keyboard.
```

File: py-rse/bash-advanced.Rmd [hunk 8]
```diff
@@ -4,15 +4,15 @@
...
 ```
 
-Chapter \@ref(py-rse-bash-basics) explained how we can use the command line
-to do all of the things we can do with a graphical file explorer,
-and how to go beyond that to combine commands in powerful ways using pipes and redirection.
-This chapter extends those ideas to show how we can save commands in files
-to create new tools of our own,
+The previous chapter explained how we can use the command line
+to do all of the things we can do with a GUI,
+and how combine commands in new ways using pipes and redirection.
```
[the remark this answers is not in the document]
Comment (gvwilson) [reply py-rse/bash-advanced.Rmd:9]: :+1:

File: py-rse/bash-advanced.Rmd [hunk 9]
```diff
@@ -4,15 +4,15 @@
...
+This chapter extends those ideas to show
+how we can create new tools by saving commands in files
 and how to use a more powerful version of [wildcards][wildcard]
 to extract data from files.
 
-We'll be continuing to work in `zipfs_law/`,
+We'll be continuing to work in the `zipf` projecct,
```
[the remark this answers is not in the document]
Comment (gvwilson) [reply py-rse/bash-advanced.Rmd:15]: :+1:

File: py-rse/bash-advanced.Rmd [hunk 10]
```diff
@@ -27,47 +27,47 @@ which after the previous chapter should contain the following files:
...
-Note that we do *not* put a dollar sign `$` at the front of the line:
-we have been showing that for interactive commands,
+Note that we do *not* put the `$` prompt at the front of the line.
+We have been showing that to highlight interactive commands,
 but in this case we are putting the command in a file rather than running it immediately.
 
-Once we have added this line to the file,
-we can write it out with <kbd>Ctrl</kbd>+<kbd>O</kbd>
+Once we have added this line,
+we can save the file with <kbd>Ctrl</kbd>+<kbd>O</kbd>
```
[the remark this answers is not in the document]
Comment (gvwilson) [reply py-rse/bash-advanced.Rmd:70]: good catch.

File: py-rse/bash-advanced.Rmd (hunk 1 above)
Comment (gvwilson) [reply py-rse/bash-advanced.Rmd:261]: I think I've fixed this - can you please check?

File: py-rse/bash-advanced.Rmd (hunk 2 above)
Comment (gvwilson) [reply py-rse/bash-advanced.Rmd:532]: oops - thanks.

File: py-rse/bash-advanced.Rmd (hunk 3 above)
Comment (gvwilson) [reply py-rse/bash-advanced.Rmd:652]: Used capital 'L' to start a new sentence rather than joining them up.

File: py-rse/bash-advanced.Rmd (hunk 4 above)
Comment (gvwilson) [reply py-rse/bash-advanced.Rmd:855]: :+1:

File: py-rse/bash-advanced.Rmd (hunk 5 above)
Comment (gvwilson) [reply py-rse/bash-advanced.Rmd:900]: Seems to be defined in the glossary?

File: py-rse/bash-advanced.Rmd (hunk 6 above)
Comment (gvwilson) [reply py-rse/bash-advanced.Rmd:910]: Fixed.

File: py-rse/bash-advanced.Rmd (hunk 7 above)
Comment (gvwilson) [reply py-rse/bash-advanced.Rmd:955]: Fixed.

**Comment by gvwilson** [comment]
Changes made - there are two FIXME's that I'd appreciate help with.

**Pull request merged**